cpp/src/Ice/Exception.cpp

```diff
@@ -531,9 +531,14 @@ Ice::Exception::ice_print(ostream& out) const
 {
     if (_file && _line > 0)
     {
-        out << _file << ':' << _line << ": ";
+        out << _file << ':' << _line << ' ';
+    }
+    out << ice_id();
```

Comment:
Note this important change: we always include the ice_id() when "printing" an exception in C++.